python/pyiceberg/catalog/__init__.py

```diff
@@ -25,6 +26,41 @@
 from pyiceberg.table.partitioning import UNPARTITIONED_PARTITION_SPEC, PartitionSpec
 from pyiceberg.table.sorting import UNSORTED_SORT_ORDER, SortOrder
 from pyiceberg.typedef import EMPTY_DICT, Identifier, Properties
+from pyiceberg.utils.config import Config, merge_config
+
+logger = logging.getLogger(__name__)
+
+_env_config = Config()
+
+
+def load_catalog(name: str, uri: str, **properties: str | None) -> Catalog:
+    from pyiceberg.catalog.rest import RestCatalog
+
+    supported_catalogs: dict[str, type[Catalog]] = {"http": RestCatalog}
+
+    try:
+        # In case Thrift isn't installed
+        from pyiceberg.catalog.hive import HiveCatalog
+
+        supported_catalogs["thrift"] = HiveCatalog
+    except ImportError:
+        logger.warning("Apache Hive not supported, to enable: pip install 'pyiceberg[hive]'")
```

Comment:
I don't think we should warn if something isn't installed. The right place to do this is when a `thrift://` URI is passed in. If we see one and Hive isn't installed, we can have a nice error message.

Reply:
Updated 👍🏻